src/commands/course/_common.ts

```diff
@@ -0,0 +1,149 @@
+import { Cursor, Collection as MongoCollection } from 'mongodb';
+import { Collection } from '../../database';
+import ExecutionContext from '../../ExecutionContext';
+import { bold } from '../../format';
+
+export type Course = {
+  _id: string;
+  subject: string;
+  number: number;
+  name: string;
+  roleId: string;
+  members: string[];
+};
+
+/**
+ * Returns a formatted version of the course name.
+ *
+ * @param course The course to format.
+ */
+export const formatCourse = (course: Course): string => {
+  return `${course._id} - ${course.name}`;
+};
+
+/**
+ * Returns a partially-bolded version of the full course string. Only the course
+ * subject and number are bolded, not the name.
+ *
+ * @param course The course to format and partially bold.
+ */
+export const semiBoldCourse = (course: Course): string => {
+  return bold(course._id) + ` - ${course.name}`;
+};
+
+/**
+ * Returns a bolded version of the full course string.
+ *
+ * @param course The course to format and bold.
+ */
+export const boldCourse = (course: Course): string => {
+  return bold(formatCourse(course));
+};
+
+export const courseExists = async (
+  ctx: ExecutionContext,
+  guildId: string,
+  id: string,
+): Promise<boolean> => {
+  return !!(await getCourseById(ctx, guildId, id));
+};
+
+export const getCourseById = async (
+  ctx: ExecutionContext,
+  guildId: string,
+  id: string,
+): Promise<Course | null> => {
+  return coursesCollection(ctx, guildId).findOne({ _id: id });
+};
+
+export const getCourseByRoleId = async (
+  ctx: ExecutionContext,
+  guildId: string,
+  roleId: string,
+): Promise<Course | null> => {
+  return coursesCollection(ctx, guildId).findOne({ roleId });
+};
+
+export const scanCourses = async (
+  ctx: ExecutionContext,
+  guildId: string,
+): Promise<Cursor<Course>> => {
+  return coursesCollection(ctx, guildId).find();
+};
+
+export const createCourse = async (
+  ctx: ExecutionContext,
+  guildId: string,
+  courseInfo: Course,
+): Promise<void> => {
+  await coursesCollection(ctx, guildId).insertOne(courseInfo);
+};
+
+export const deleteCourse = async (
+  ctx: ExecutionContext,
+  guildId: string,
+  courseInfo: Course,
+): Promise<void> => {
+  const db = await ctx.db.getDb(guildId);
+
+  await db
+    .collection(Collection.COURSES)
+    .deleteOne({ roleId: courseInfo.roleId });
```

Comment:
Can `coursesCollection()` be used here instead?
```suggestion
  coursesCollection(ctx, guildId).deleteOne({ roleId: courseInfo.roleId }));
```